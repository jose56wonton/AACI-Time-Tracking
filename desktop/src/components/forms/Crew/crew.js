import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Grid, Typography, Button } from '@material-ui/core';
import { Field, Form } from 'formik';
import { withStyles } from '@material-ui/core/styles';
import TextField from 'components/inputs/TextField';
import styles from './styles';

class Crew extends Component {
  render() {
    const { classes, errors, initialValues, resetForm, isSubmitting, label } = this.props;
    return (
      <Form>
        <Grid container spacing={24} className={classes.gridContainer}>
          <Grid item xs={12} className={classes.row}>
            <Typography variant="h6">{label}</Typography>
          </Grid>
          <Grid item xs={12} className={classes.row}>
            <Field
              name="name"
              component={TextField}
              margin="none"
              label="Name"
              type="search"
              className={classes.field}
              helper="normal"
            />
          </Grid>
          <Grid item xs={12} className={classes.row}>
            <Typography
              color="error"
              variant="button"
              className={classes.field}
            >
              {errors.submit}
            </Typography>
            <div>
              <Button
                type="submit"
                color="primary"
                disabled={isSubmitting}
                variant="contained"
                className={classes.button}
              >
                Save
              </Button>
              <Button
                onClick={() => {
                  resetForm(initialValues);
                }}
                disabled={isSubmitting}
                color="secondary"
                variant="text"
                className={classes.button}
              >
                Reset
              </Button>
            </div>
          </Grid>
        </Grid>
      </Form>
    );
  }
}

Crew.propTypes = {
  classes: PropTypes.object.isRequired,
  label: PropTypes.string.isRequired,
  isSubmitting: PropTypes.bool.isRequired,
  resetForm: PropTypes.func.isRequired,
  initialValues: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired
};


export default withStyles(styles)(Crew);