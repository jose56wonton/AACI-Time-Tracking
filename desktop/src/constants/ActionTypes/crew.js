export const GET_CREWS_REQUEST = 'get_crews_request';
export const GET_CREWS_SUCCESS = 'get_crews_success';
export const GET_CREWS_FAILURE = 'get_crews_failure';

export const EDIT_CREWS_MODAL = 'edit_crews_modal';

export const UPDATE_CREW_REQUEST = 'update_crew_request';
export const UPDATE_CREW_SUCCESS = 'update_crew_success';
export const UPDATE_CREW_FAILURE = 'update_crew_failure';