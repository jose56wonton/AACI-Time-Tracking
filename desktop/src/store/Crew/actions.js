import { crewActionTypes } from 'constants/actionTypeConstants';
import { modalActions,genericActions,snackActions } from 'store/actions';
import domains from 'constants/domains';
import * as status from 'constants/status';

export const getAllCrews = () => {
  return async dispatch => {
    dispatch(genericActions.getAll(domains.CREW));    
  };
};

export const editCrewsModal = () => {
  return  modalActions.openModal(crewActionTypes.EDIT_CREWS_MODAL, null);
};

export const updateCrew = crew => {
  return async dispatch => {
    dispatch({ type: crewActionTypes.UPDATE_CREW_REQUEST });
    try {
      await dispatch(genericActions.put(domains.CREW,crew));
      await dispatch(snackActions.openSnack(status.SUCCESS, 'Crew Updated'));
      return dispatch({ type: crewActionTypes.UPDATE_CREW_SUCCESS });      
    } catch (e) {
      console.log(e);
      await dispatch(snackActions.openSnack(status.SUCCESS, 'Crew Update Failed'));
      return dispatch({ type: crewActionTypes.UPDATE_CREW_FAILURE });
    }
  };
};
