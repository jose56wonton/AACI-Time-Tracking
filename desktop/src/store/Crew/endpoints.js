import { HOST } from 'constants/network';

import axios from 'axios';
const DOMAIN = 'crews';

export const getCrews = () => {
  return axios.get(`${HOST}/${DOMAIN}`);
};
export const putCrew = (id,crew) => {
  return axios.put(`${HOST}/${DOMAIN}/${id}`, { ...crew });
};
