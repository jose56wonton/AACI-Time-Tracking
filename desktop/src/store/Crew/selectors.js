import { createSelector } from 'reselect';

export const getCrewsFromEntities = state => state.entities.crews;
export const getCrewsFromResults = state => state.results.crews;

export const getAnalyzeState = state => state.analyze;

export const getAllCrews = createSelector(
  getCrewsFromEntities,
  getCrewsFromResults,
  (crews, results) => {
    if (!results || results.size === 0) return null;
    return results.map(crewId => {
      return crews[crewId];
    });
  },
);


export const getSelectedCrew = createSelector(
  getCrewsFromEntities,
  getAnalyzeState,
  (crews, analyze) => {
    if(analyze.crew === -1) 
      return {};
    else 
      return crews[analyze.crew];
  }
);