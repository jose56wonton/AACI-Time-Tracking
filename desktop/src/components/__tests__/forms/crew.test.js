import React from 'react';
import { shallow } from 'enzyme';

import { Crew } from 'components/forms/Crew/crew';
import CrewHOC from 'components/forms/Crew';

const props =  {  
  classes: {},
  label: 'label',
  isSubmitting: true,
  resetForm: jest.fn(),
  initialValues: {},
  errors: {}
};


const setup = overRides => {  
  return shallow(<Crew {...props} {...overRides}/>);    
};

const setupHOC = overRides => {
  return shallow(<CrewHOC {...props} {...overRides}/>);
};

describe.skip('Crew Component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();    
  });
  it('should render correctly withStyles', () => {
    const wrapper = setupHOC();
    expect(wrapper).toMatchSnapshot();
  });
  it('should call resetForm on crew-reset-button', () => {
    const wrapper = setup();
    const instance = wrapper.instance();
    instance.resetForm = jest.fn();
    expect(props.resetForm).toHaveBeenCalledTimes(0);
    wrapper.find('#crew-reset-button').first().simulate('click');
    expect(props.resetForm).toHaveBeenCalledTimes(1);
  });
});