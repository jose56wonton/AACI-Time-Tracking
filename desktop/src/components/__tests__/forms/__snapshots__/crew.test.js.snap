// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crew Component should render correctly 1`] = `
<Form>
  <WithStyles(Grid)
    container={true}
    spacing={24}
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <WithStyles(Typography)
        variant="h6"
      >
        label
      </WithStyles(Typography)>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <C
        component={[Function]}
        helper="normal"
        label="Name"
        margin="none"
        name="name"
        type="search"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <WithStyles(Typography)
        color="error"
        variant="button"
      />
      <div>
        <WithStyles(Button)
          color="primary"
          disabled={true}
          type="submit"
          variant="contained"
        >
          Save
        </WithStyles(Button)>
        <WithStyles(Button)
          color="secondary"
          disabled={true}
          id="crew-reset-button"
          onClick={[Function]}
          variant="text"
        >
          Reset
        </WithStyles(Button)>
      </div>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</Form>
`;

exports[`Crew Component should render correctly withStyles 1`] = `
<Crew
  classes={
    Object {
      "button": "Crew-button-7",
      "error": "Crew-error-6",
      "field": "Crew-field-4",
      "gridContainer": "Crew-gridContainer-1",
      "headerRow": "Crew-headerRow-3",
      "row": "Crew-row-2",
      "switchBox": "Crew-switchBox-5",
    }
  }
  errors={Object {}}
  initialValues={Object {}}
  isSubmitting={true}
  label="label"
  resetForm={[Function]}
/>
`;
